src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -328,14 +329,14 @@ internal MethodSignatureBase GetEffectiveCtor()
             return span.Slice(start, end - start).Trim().ToString();
         }
 
-        private bool IsParamMatch(IReadOnlyList<Parameter> methodParameters, ImmutableArray<TypedConstant> suppressionParameters)
+        private bool IsParamMatch(IReadOnlyList<Parameter> methodParameters, INamedTypeSymbol[] suppressionParameters)
         {
             if (methodParameters.Count != suppressionParameters.Length)
                 return false;
 
             for (int i = 0; i < methodParameters.Count; i++)
             {
-                var namedSymbol = ((INamedTypeSymbol)suppressionParameters[i].Value!);
+                var namedSymbol = suppressionParameters[i];
                 var suppressionParamType = namedSymbol.GetCSharpType(_typeFactory);
```

Comment:
We have a method to assert if a symbol and a CSharpType are the same type: https://github.com/Azure/autorest.csharp/blob/4fd31dcd869853538d7f47f6cc98b2fe5b022c3a/src/AutoRest.CSharp/Common/Utilities/NamedTypeSymbolExtensions.cs#L35